scripts/process_uploads.py

```diff
@@ -65,15 +65,51 @@ def determine_samples(
 
 def generate_file_list(samples: List[str]):
     """ Generate list of expected files, given a list of sampleIDs """
-    main_files: List[str] = []
-    archive_files: List[str] = []
+    main_files = []
+    archive_files = []
     for s in samples:
-        main_files.extend([f'{s}.g.vcf.gz', f'{s}.g.vcf.gz.tbi', f'{s}.g.vcf.gz.md5'])
-        archive_files.extend([f'{s}.cram', f'{s}.cram.crai', f'{s}.cram.md5'])
+        sample_group_main = [f'{s}.g.vcf.gz', f'{s}.g.vcf.gz.tbi', f'{s}.g.vcf.gz.md5']
+        main_files.append(sample_group_main)
+        sample_group_archive = [f'{s}.cram', f'{s}.cram.crai', f'{s}.cram.md5']
+        archive_files.append(sample_group_archive)
 
     return main_files, archive_files
 
 
+def setup_job(batch: hb.batch, name: str, docker_image: Optional[str]):
+    """ Define a new hail batch job """
+
+    new_j = batch.new_job(name=name)
+    if docker_image is not None:
+        new_j.image(docker_image)
+
+    new_j.command(
+        'gcloud -q auth activate-service-account --key-file=/gsa-key/key.json'
+    )
+
+    return new_j
+
+
+def validate_md5(job: hb.batch.job, sample_group: List[str], upload_path: str):
+    """Each gvcf and cram file is uploaded with a corresponding md5 checksum.
+    This function validates each .md5 file, by recalculating the md5 checksum
+    and comparing this with the uploaded .md5."""
+
+    # Generate paths to files that are being validated
+    sample_gvcf = sample_group[0]
+    md5_file = sample_group[2]
+    path_to_gvcf = join('gs://', upload_path, sample_gvcf)
+    path_to_md5 = join('gs://', upload_path, md5_file)
+
+    # Calculate md5 checksum.
+    job.command(
+        f'gsutil cat {path_to_gvcf} | md5sum | sed "s/-/{sample_gvcf}/" > {job.ofile}'
```

Comment:
`job.ofile` is useful for delocalizing, but we're not really using the output afterwards, right? Should this just to a temporary file?

Reply:
Have modified the implementation, let me know if that's better?